File: src/main/java/br/anhembi/funmodechild/service/CategoryService.java
package br.anhembi.funmodechild.service;

import br.anhembi.funmodechild.entity.Category;
import br.anhembi.funmodechild.model.response.CategoryResponse;
import br.anhembi.funmodechild.repository.CategoryRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CategoryService {

    private final CategoryRepository categoryRepository;

    public CategoryService(CategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }

    public List<CategoryResponse> getCategories() {
        return categoryRepository.findAll()
            .stream().map(Category::toApiResponse)
            .toList();
    }
}

File: src/main/kotlin/br/anhembi/funmodechild/service/CategoryService.kt
package br.anhembi.funmodechild.service

import br.anhembi.funmodechild.model.response.CategoryResponse
import br.anhembi.funmodechild.repository.CategoryRepository
import org.springframework.stereotype.Service

@Service
class CategoryService(private val categoryRepository: CategoryRepository) {
    fun listCategories(): List<CategoryResponse> =
        categoryRepository.findAll()
            .map { it.toApiResponse() }
}
